presto-tests/src/test/java/com/facebook/presto/tests/TestLocalBinarySpilledQueries.java

```diff
@@ -102,4 +103,212 @@ public void testRowFieldAccessorInAggregate()
     {
         super.testRowFieldAccessorInAggregate();
     }
+
+    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "Spilling for RIGHT and FULL outer joins is unsupported")
```

Comment:
generally I don't like `expectedException` to be used in that way. See `com.facebook.presto.tests.AbstractTestQueryFramework#assertQueryFails`. With similar method you could test all the possibilities in one method and it would much more readable.